src/routing.js

```diff
@@ -165,14 +165,16 @@ export function LocationProvider({
 
 export function StaticLocationProvider({ location, children }) {
   if (!location) throw new Error(`Your need to supply a location for the static location provider`)
+  const { location: locationLive } = React.useContext(locationAndLocationMatchContext)
 
-  const navigate = React.useCallback(
+  const navigateStatic = React.useCallback(
```

Comment:
I think this can possibly lead to errors where, technically, the developer didn't do anything wrong. For instance, when you use this libraries Link component to navigate around, the user might click a link in an element that's transitioning out, right? 

We could change this behavior to just show a warning in development, but on the other hand you might actually want to throw an error if the developer calls navigate himself. Maybe we should add the `location === locationLive` check to the `Link` component as well, changing it so it doesn't do anything while transitioning?

Reply:
Good point. The `Link` option is in my opinion not viable as the same argument is valid for buttons. I think the `navigate` function would indeed be the correct place to adjust the behavior.

I however am not sure we want to deal with this potential problem now. We seem to have 2 options:

1. Figure out the best behavior without any use cases and implement it
2. Keep the error and fix the problem when it arises

I'm inclined to go for option 2. Throwing an error will not break the application if a user triggers a call to navigate, that trigger is an event handler and thus will not harm any rendering. During development it will force developers to fix a potentially very harmful problem (a context that prevents navigation).

Reply:
Yeah, let's not go with option 1. Maybe share this info in our #technology channel upon merge, so people have seen this before they run into it.